sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -104,3 +106,25 @@ def test_check_estimators_unfitted():
     # check that CorrectNotFittedError inherit from either ValueError
     # or AttributeError
     check_estimators_unfitted("estimator", CorrectNotFittedErrorClassifier)
+
+
+def test_common():
+    try:
+        import pandas as pd
+    except ImportError:
+        raise SkipTest("Pandas not found")
+    X = pd.DataFrame([[1, 1], [1, 2], [1, 3], [2, 1], [2, 2], [2, 3]])
+    y = pd.Series([1, 1, 1, 2, 2, 2])
+    weights = pd.Series([1,1,1,1,1,1])
+    for name, Estimator in all_estimators():
+        estimator = Estimator()
+        if 'sample_weight' in inspect.getargspec(estimator.fit)[0]:
```

Comment:
we have a helper `sklearn.utils.validation.has_fit_parameter` which might help here